File: EduHubConnect-Backend-UsingSpringBoot/src/main/java/com/connect/eduHub/model/FriendRequest.java
package com.connect.eduHub.model;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(
	    name = "friend_requests",
	    uniqueConstraints = @UniqueConstraint(columnNames = {"sender_id", "receiver_id", "status"})
	)

public class FriendRequest {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private User sender;

    @ManyToOne
    private User receiver;

    @Enumerated(EnumType.STRING)
    private RequestStatus status = RequestStatus.PENDING; // PENDING, ACCEPTED, DECLINED

    private LocalDateTime sentAt;
}

File: EduHubConnect-AndroidFrontEnd/app/src/main/java/com/connect/eduhubconnect/model/FriendRequest.kt
package com.connect.eduhubconnect.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "friend_requests")
data class FriendRequest(
    @PrimaryKey val id: Long,
    val senderId: Long,
    val receiverId: Long,
    val status: String,
    val sentAt: String,
    val senderName: String,
    val userBio: String,
    val senderEmail: String,
    val senderProfileImage: String
)
